addons/website_cf_turnstile/static/src/js/turnstile.js

```diff
@@ -61,9 +74,50 @@ export const turnStile = {
         this._super(...arguments);
     },
 
-    addSpinner(button) {
+    /**
+     * Take the result of `addTurnstile` and render it if needed
+     */
+    renderTurnstile: function (turnstileNodes) {
+        turnstileNodes = turnstileNodes.toArray();
+        const turnstileContainer = turnstileNodes.filter((node) =>
+            node.classList.contains("s_turnstile_container")
+        )[0];
+        const turnstileScript = turnstileNodes.filter(
+            (node) => node.id === "s_turnstile_remote_script"
+        )[0];
+        // there should only be a remote script if it was loaded for the first time
+        if (turnstileScript) {
+            return;
+        }
+        if (
+            window.turnstile?.render &&
+            turnstileContainer &&
+            !turnstileContainer.querySelector("iframe")
+        ) {
+            window.turnstile.render(turnstileContainer);
+        }
+    },
+
+    _createSpinner() {
         const spinner = document.createElement("i");
         spinner.classList.add("fa", "fa-refresh", "fa-spin", "turnstile-spinner");
+        return spinner;
+    },
+
+    /**
+     * same as addSpinner but does not set innerText
+     */
+    addSpinnerNoMangle(button) {
+        const spinner = this._createSpinner();
+        spinner.classList.add("me-1");
+        button.disabled = true;
+        button.classList.add("disabled");
+        button.prepend(spinner);
+    },
```

Comment:
Fixed
This thing breaks on runbot but not locally so it was fun :smiling_face_with_tear:
Probably an override of event adds line breaks that are normally invisible but innerText belows transforms them into <br> which are very visible